clevertap-core/src/main/java/com/clevertap/android/sdk/CleverTapAPI.java

```diff
@@ -1637,25 +1638,36 @@ public void setInAppNotificationListener(InAppNotificationListener inAppNotifica
     }
 
     /**
-     * Returns the PushPermissionNotificationResponseListener object
+     * This method unregisters the given instance of the PushPermissionResponseListener if
+     * previously registered.
+     * <p>
+     * Use this method to stop observing the push permission result.
      *
-     * @return An {@link PushPermissionResponseListener} object
+     * @param pushPermissionResponseListener An {@link PushPermissionResponseListener} object
      */
-    @SuppressWarnings({"unused", "WeakerAccess"})
-    public PushPermissionResponseListener getPushPermissionNotificationResponseListener() {
-        return coreState.getCallbackManager().getPushPermissionResponseListener();
+    @SuppressWarnings({"unused"})
+    public void unregisterPushPermissionNotificationResponseListener(PushPermissionResponseListener
+                                                                           pushPermissionResponseListener) {
+        coreState.getCallbackManager().
+                unregisterPushPermissionResponseListener(pushPermissionResponseListener);
     }
 
     /**
-     * This method sets the PushPermissionNotificationResponseListener
+     * This method registers the PushPermissionNotificationResponseListener.
+     * <p>
+     * Make sure to unregister the listener in onDestroy() or onPause() of the activity/fragment if the
+     * listener is registered in onCreate() or onResume() respectively.
```

Comment:
@shivamsharma2710 Please mention to call this register method only in `onCreate`